src/components/OcButton.spec.js

```diff
@@ -0,0 +1,146 @@
+import OcButton from "./OcButton.vue";
+import {createLocalVue, mount, shallowMount} from "@vue/test-utils";
+
+const localVue = createLocalVue()
+
+
+describe("OcButton", () => {
+  const testSlots = { default: '<p class="text">Test button</p>' }
+  const route = {
+    path: '/some/path',
+    name: 'some-route'
+  }
+  function getWrapperWithProps(props) {
+    return shallowMount(OcButton, { localVue: localVue, propsData: props })
+  }
+  function getWrapperWithTestSlot() {
+    return shallowMount(OcButton, { localVue: localVue, slots: testSlots })
+  }
+  function getWrapperWithPropsAndMockRouter(props) {
+    return mount(OcButton, {
+      localVue: localVue,
+      slots: testSlots,
+      propsData: props,
+      mocks: {
+        $route: route
+      }
+    })
+  }
+
+  it("displays slot html", () => {
+    const wrapper = getWrapperWithTestSlot()
+    const slot = wrapper.find('p')
+    expect(slot).toBeTruthy()
+    expect(slot.attributes("class")).toBe("text")
+    expect(slot.text()).toBe("Test button")
+  })
+
+  it('can stop class propagation', () => {
+    const wrapper = getWrapperWithProps({
+      stopClassPropagation: true,
+      size: 'middle'
+    })
+    expect(wrapper.attributes("class")).toContain("")
+  })
+
+  describe("emit click event", () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapperWithProps({})
+    })
+    it("emits click event when click is triggered", async () => {
+      await wrapper.trigger("click")
+      await wrapper.vm.$nextTick()
+      expect(wrapper.emitted('click')).toBeTruthy()
+    })
+    it("does not emit click event when input is triggered", async () => {
+      await wrapper.trigger("input")
+      await wrapper.vm.$nextTick()
+      expect(wrapper.emitted('click')).toBeFalsy()
+    })
+  })
+  describe('disabled', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapperWithProps({ disabled: true })
+    })
+    it('should have disabled attribute set to disabled', () => {
+      expect(wrapper.attributes("disabled")).toBe("disabled")
+    })
+    it("does not emit click event", async () => {
+      await wrapper.trigger("click")
+      await wrapper.vm.$nextTick()
+      expect(wrapper.emitted('click')).toBeFalsy()
+    })
+    it("should not route to provided router link", async () => {
+      wrapper = getWrapperWithPropsAndMockRouter({
+        type: "router-link",
+        to: {
+          name: 'some-route'
```

Comment:
if you use `route.path` in other places you could also use `route.name` here
or you hard-code it everywhere. As this is a property that is valid only for this single test a hard-coded value might make it more readable. And changing all routes in the whole test-file should not be ever needed

Reply:
it is used in two places, so I've used `route.path` to avoid duplications

Reply:
i've removed the tests to check for routes because they're not working properly.  for this scenario, while prop `disable: true`, the route should not be updated but the current configuration is not catching that. I'll add route tests separately after some research